### PR TITLE
Make tfcompile more friendly on other archs.

### PR DESCRIPTION
**Pull request by adpopescu**
Currently tfcompile and tfcompile_wrapper use hard coded values for target_triple. The hard coded value is `x86_64-pc-linux`. This makes the following test(s) fail on s390x:

//tensorflow/python/tools:saved_model_cli_test

I've removed the hard coded string and changed it so that the default value from llvm is used instead.

In the case where a specific target_triple is desired such as in the `testAOTCompileCPUFreezesAndCompilesVariablesToFeedNoneTargetAarch64Linux` and `testAOTCompileCPUFreezesAndCompilesVariablesToFeedNoneTargetAarch64Android` subtests, the set value is still allowed to follow through.

The subtests above cause additional failures on big endian systems due to a check for little endianness. I added an exception to skip these two subtests on s390x.

tfcompile_wrapper doesn't set a default target_cpu. This is causing the following test case to fail on s390x:

//tensorflow/python/tools:aot_compiled_test

I've made a similar change where I use the value from llvm as the default.

### CODE REVIEW

File: tensorflow/python/tools/saved_model_cli.py [hunk 1]
```diff
@@ -184,7 +184,7 @@
     ' object.')
 
 _SMCLI_TARGET_TRIPLE = flags.DEFINE_string(
-    name='target_triple', default='x86_64-pc-linux',
+    name='target_triple', default='',
     help='Triple identifying a target variation, containing information such as'
     ' processor architecture, vendor, operating system, and environment. '
     'Defaults to \'x86_64-pc-linux\'.')
```
Comment (BlaziusMaximus) [inline tensorflow/python/tools/saved_model_cli.py:190]: Update help string to reflect new default value.

Comment (adpopescu) [reply tensorflow/python/tools/saved_model_cli.py]: Thank you. Should be fixed now. Sorry about that.

Comment (BlaziusMaximus) [reply tensorflow/python/tools/saved_model_cli.py]: All good. Thanks for the change!

**Review by BlaziusMaximus — APPROVED** [review]
LGTM

**Review by sagunb — APPROVED** [review]

**Review by sagunb — APPROVED** [review]

**Review by sagunb — APPROVED** [review]

**Pull request merged**